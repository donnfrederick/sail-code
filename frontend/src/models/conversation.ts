import { Questionnaire } from 'models/evaluations'
import { UserProfile as User } from 'models/userProfile'

export type TimeSelectType = 'from' | 'zone'

export type ConversationMode = 'soundOnly' | 'video'

export type ReservationType = 'anyone' | 'request'

export interface Response {
  channel_id: string
  chats: any[]
  id: number
  status: string
  start_at: string
  end_at: string
  created_at: string
  updated_at: string
  users: any[]
}

export interface ConversationRequestPostParams {
  reservable_conversation_id: number
  start_at: string
}

export interface Request {
  accepting_requests: number
  start_at: string
  end_at: string
}

export interface ConversationRequest {
  start_on?: string
  end_on?: string
  term?: 'week' | 'month'
  page?: number
}

export interface UserConversation {
  accepting_requests: boolean
  channel_id: string
  chats: any[]
  id: number
  status: string
  statuses: ConversationStatus[]
  start_at: string
  end_at: string
  evaluate: ConversationRate<Evaluate>[]
  memos: ConversationRate<Memo>[]
  reports: ConversationRate<Report>[]
  created_at: string
  updated_at: string
  with_self: boolean
}

export interface Chats {
  content: string
  conversation_id: number
  created_at: string
  user_picture_url: string
  id: number
  updated_at: string
  user_id: number
}

export interface Conversation {
  accepting_requests: boolean
  available: boolean
  channel_id: string
  chats: Chats[]
  id: number
  status: string
  statuses: ConversationStatus[]
  start_at: string
  end_at: string
  evaluate: ConversationRate<Evaluate>[]
  memos: ConversationRate<Memo>[]
  reports: ConversationRate<Report>[]
  created_at: string
  updated_at: string
  users: User[]
  conversation_requests: RequestConversationInConversation[]
}

export interface RequestConversation {
  conversation: Conversation
  end_at: string
  id: number
  start_at: string
  user: User
}

export interface RequestConversationInConversation {
  end_at: string
  id: number
  start_at: string
  user: User
}

export interface CancelledConversation {
  conversation_id: number
  start_at: string
  end_at: string
  created_at: string
  updated_at: string
  users: User[]
}

export interface Meta {
  start_on: string
  end_on: string
}

export interface ServerTime {
  current_time: string
  current_time_epoch: number
  end_at: string
  end_at_epoch: number
}

export interface ConversationResponse {
  data: Conversation[]
  meta: Meta
}

export interface RequestConversationResponse {
  data: RequestConversation[]
}

export interface Calendar {
  [date: string]: {
    is_enabled: boolean
    is_reserved: boolean
  }
}

export interface Topics {
  [key: string]: string
}

export type ConversationStatusValue = 'Ontime' | 'Late' | 'Absent'

export interface ConversationStatus {
  user_id: number
  status: ConversationStatusValue
}

export type ConversationRate<T> = { [P in keyof T]: T[P] } & { user_id: number }

interface Memo {
  memo: string
}
interface Evaluate {
  evaluate: Questionnaire
}

interface Report {
  report: ReportContentRequest
}
export type ReportReasonsUnion = 1 | 2 | 3 | 4 | 5

export interface ReportContentRequest {
  report_detail: string
  report_reasons: ReportReasonsUnion[]
}

export enum TopicEnumJa {
  '料理',
  '読書',
  'スポーツ',
  '歴史',
  '音楽',
  '芸術',
  '哲学',
  '旅行',
  '社会'
}

export enum TopicEnumEn {
  'Cooking',
  'Reading',
  'Sports',
  'History',
  'Music',
  'Art',
  'Philosophy',
  'Travel',
  'Society'
}
