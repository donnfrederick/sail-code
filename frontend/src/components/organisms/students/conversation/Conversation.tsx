import ConversationBalloon from 'components/atoms/ConversationBalloon'
import RoundImage from 'components/atoms/RoundImage'
import NetworkStatus from 'components/atoms/students/NetworkStatus'
import RemainingTime from 'components/atoms/students/RemainingTime'
import ConversationButtons from 'components/molecules/students/ConversationButtons'
import CloseConversationModal from 'components/organisms/students/modal_contents/close_conversation'
import withConversationId from 'hocs/withConversationId'
import * as ConversationModels from 'models/conversation'
import * as React from 'react'
import styled from 'styled-components'
import getPartner from 'utils/getPartner'
import resolvePath from 'utils/resolvePath'
import * as SoraClient from 'utils/SoraClient'
import Chat from 'components/molecules/Chat'

interface Props {
  bandwidth: number
  chats: ConversationModels.Chats[]
  closeSoon: boolean
  conversation: ConversationModels.Conversation | null
  conversationMode: ConversationModels.ConversationMode
  currentTopic: string
  isConnecting: boolean
  myId: number
  remainingTime: string
  topics: ConversationModels.Topics
  youAreLate: boolean
  openChat(): void
  openModal(): void
  setModalContents(contents: JSX.Element): void
}

export default withConversationId((props: Props) => {
  const {
    bandwidth,
    chats,
    closeSoon,
    conversation,
    conversationMode,
    // currentTopic,
    isConnecting,
    myId,
    remainingTime,
    // topics,
    youAreLate,
    openModal,
    setModalContents
  } = props

  if (conversation) {
    SoraClient.init(conversation.channel_id)
    SoraClient.connect()
  }

  const [isChatOpen, setIsChatOpen] = React.useState(false)
  const partner = conversation ? getPartner(conversation, myId) : null

  return (
    <Container>
      <RemoteVideoContainer>
        <RemoteVideo id="remote-video" autoPlay={true} playsInline={true} />
      </RemoteVideoContainer>
      <LocalVideoContainer>
        <LocalVideo id="local-video" autoPlay={true} playsInline={true} />
      </LocalVideoContainer>
      <NetworkStatus
        status={bandwidth > 300 ? 'good' : bandwidth > 100 ? 'normal' : 'bad'}
      />
      <RemainingTime remainingTime={remainingTime} />
      {conversationMode === 'soundOnly' ? (
        <Partner>
          <RoundImage
            src={
              partner
                ? partner.picture_url
                : resolvePath.image('common/user.png')
            }
            size={200}
            badge={partner ? (partner.highly_reliable ? 'elder' : null) : null}
            code={partner ? partner.country_code : ''}
          />
          <PartnerName>{partner ? partner.name : null}</PartnerName>
        </Partner>
      ) : null}
      {/* {isChatOpen ? <Topics topics={topics} /> : null}
      {!isChatOpen && currentTopic ? (
        <ConversationBalloon
          text={translateTopicToEnglish(currentTopic)}
          isTopic={true}
        />
      ) : null} */}
      {closeSoon ? (
        <ConversationBalloon text="It is time to say goodbye" />
      ) : null}
      {youAreLate ? (
        <ConversationBalloon text="You are late！" color="#ff676e" />
      ) : null}
      {isConnecting ? <Connecting>{'Connecting...'}</Connecting> : null}
      <ConversationButtons
        conversationMode={conversationMode}
        isChatOpen={isChatOpen}
        openChat={() => setIsChatOpen(!isChatOpen)}
        closeEvent={() => {
          setModalContents(<CloseConversationModal />)
          openModal()
        }}
      />
      <Canvas id="check-stream" />
      {conversation &&
        isChatOpen && (
          <ChatContainer>
            <Chat
              chats={chats}
              conversationId={conversation.id}
              openChat={() => setIsChatOpen(!isChatOpen)}
            />
          </ChatContainer>
        )}
    </Container>
  )
})

const Container = styled.div`
  position: relative;
  width: 100%;
  height: 100%;
  background-color: #000;
`

const LocalVideoContainer = styled.div`
  position: absolute;
  top: 32px;
  right: 32px;
  width: 148px;
  height: 238px;
  border-radius: 8px;
  overflow: hidden;
`

const LocalVideo = styled.video`
  position: absolute;
  top: 50%;
  left: 50%;
  height: 100%;
  transform: translate(-50%, -50%);
`

const RemoteVideoContainer = styled.div`
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 80%;
  overflow: hidden;
`

const RemoteVideo = styled.video`
  position: absolute;
  top: 50%;
  left: 50%;
  height: 100%;
  transform: translate(-50%, -50%);
`

const Partner = styled.div`
  position: absolute;
  top: 410px;
  left: 0;
  width: 100%;
  text-align: center;
`

const PartnerName = styled.div`
  margin-top: 32px;
  font-size: 48px;
  font-weight: 500;
  letter-spacing: 0.5px;
  text-align: center;
  color: #ffffff;
`

const Connecting = styled.div`
  position: absolute;
  top: 280px;
  right: 0;
  left: 0;
  width: 200px;
  height: 32px;
  margin: auto;
  font-size: 32px;
  color: #fff;
`

const Canvas = styled.canvas`
  display: none;
  width: 100%;
  height: 100%;
`

const ChatContainer = styled.div`
  position: absolute;
  background-color: #fff;
  bottom: 0;
  left: 0;
  width: 100%;
  height: 50%;
  overflow: hidden;
`
// visibility: ${isChatOpen ? 'visible' : 'hidden'};
