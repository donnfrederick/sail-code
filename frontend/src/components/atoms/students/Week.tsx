import * as ConversationModels from 'models/conversation'
import moment from 'moment'
import * as React from 'react'
import styled from 'styled-components'
import calendar, { getDays } from 'utils/calendar'

interface Props {
  reservations: ConversationModels.ConversationResponse
  start: string
}

export default (props: Props) => {
  const { reservations, start } = props
  return (
    <Container>
      <YearAndMonth>{getYearAndMonth(start)}</YearAndMonth>
      <Days>{getDays('en').map(day => <Day key={day}>{day}</Day>)}</Days>
      <Dates>
        {getDates(start).map((date: number, index: number) => (
          <DateElement key={date}>
            {date}
            {reservations &&
            conversationsOnDate(reservations.data, date).length ? (
              <Dot
                data-request={
                  !conversationsOnDate(reservations.data, date).find(
                    conversation => !conversation.accepting_requests
                  )
                }
              />
            ) : null}
          </DateElement>
        ))}
      </Dates>
    </Container>
  )
}

const conversationsOnDate = (
  conversations: ConversationModels.Conversation[],
  date: number
) => {
  const filtered = conversations.filter(conversation => {
    const start = moment.parseZone(conversation.start_at)
    const filteredConversationStatus =
      conversation.status === 'queued' || conversation.accepting_requests
    return start.date() === date && filteredConversationStatus
  })

  return filtered
}

const getYearAndMonth = (datestring: string) => {
  return moment.parseZone(datestring).format('MMMM YYYY')
}

const getDates = (startdate: string) => {
  const date = moment.parseZone(startdate)
  const weeks = calendar(date.year())(date.month() + 1)

  const currentWeek = weeks
    ? weeks.filter(week => {
        return week.includes(Number(date.format('D')))
      })[0]
    : [0]

  return fillout(currentWeek)
}

const fillout = (weekArray: number[], date: number = 1): any => {
  const index = weekArray.indexOf(0)
  if (index === -1) {
    return weekArray
  } else {
    weekArray[index] = date
    return fillout(weekArray, date + 1)
  }
}

const Container = styled.div`
  width: 100%;
  text-align: center;
`

const YearAndMonth = styled.div`
  margin-bottom: 24px;
  font-size: 40px;
  font-weight: 500;
  letter-spacing: 0.5px;
  text-align: center;
  color: #405766;
`

const Days = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100%;
  height: 80px;
  box-sizing: border-box;
  border-bottom: 1px solid #cad5db;
`

const Day = styled.div`
  width: 60px;
  margin: 0 13.5px;
  font-size: 28px;
  font-weight: 500;
  text-align: center;
  color: #405766;
`

const Dates = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100%;
  height: 110px;
  box-sizing: border-box;
`

const DateElement = styled.div`
  position: relative;
  top: -12px;
  width: 44px;
  margin: 0 22px;
  font-size: 36px;
  font-weight: 500;
  letter-spacing: 0.5px;
  text-align: center;
  color: #405766;
`

const Dot = styled.div`
  position: absolute;
  bottom: -27px;
  right: 0;
  left: 0;
  width: 16px;
  height: 16px;
  margin: auto;
  border-radius: 50%;
  &[data-request='true'] {
    background-color: #a3a3a3;
  }
  &[data-request='false'] {
    background-color: #1a97fd;
  }
`
