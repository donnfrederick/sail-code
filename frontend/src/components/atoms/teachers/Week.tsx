import * as ConversationModels from 'models/conversation'
import moment from 'moment'
import * as React from 'react'
import styled from 'styled-components'
import calendar, { getDays } from 'utils/calendar'

interface Props {
  reservations: ConversationModels.ConversationResponse | null
  start: string
}

export default (props: Props) => {
  const { reservations, start } = props
  return (
    <Container>
      <YearAndMonth>{getYearAndMonth(start)}</YearAndMonth>
      <Days>{getDays('ja').map(day => <Day key={day}>{day}</Day>)}</Days>
      <Dates>
        {getDates(start).map((date: number, index: number) => (
          <DateElement key={date}>
            {date}
            {reservations &&
            queuedConversationsOnDate(reservations.data)(date).length ? (
              <ReservedDot />
            ) : reservations &&
            waitingConversationsOnDate(reservations.data)(date).length ? (
              <WaitingDot
                data-request={
                  !waitingConversationsOnDate(reservations.data)(date).find(
                    conversation => !conversation.accepting_requests
                  )
                }
              />
            ) : null}
          </DateElement>
        ))}
      </Dates>
    </Container>
  )
}

type StatusType = 'queued' | 'waiting'

const conversationOnDateWithEvent = (statusType: StatusType) => (
  conversations: ConversationModels.Conversation[]
) => (date: number) => {
  const filtered = conversations.filter(conversation => {
    const start = moment.parseZone(conversation.start_at)
    return start.date() === date && conversation.status === statusType
  })

  return filtered
}

const queuedConversationsOnDate = conversationOnDateWithEvent('queued')

const waitingConversationsOnDate = conversationOnDateWithEvent('waiting')

const getYearAndMonth = (datestring: string) => {
  return moment.parseZone(datestring).format('YYYY年MM月')
}

const getDates = (startdate: string) => {
  const date = moment.parseZone(startdate)
  const weeks = calendar(date.year())(date.month() + 1)

  const currentWeek = weeks
    ? weeks.filter(week => {
        return week.includes(Number(date.format('D')))
      })[0]
    : [0]

  const filloutWeek = fillout(currentWeek)
  // 謎の複数次元配列バグを避けるため
  return removeDuplicate(filloutWeek)
}

const fillout = (weekArray: number[], date: number = 1): any => {
  const index = weekArray.indexOf(0)
  if (index === -1) {
    return weekArray
  } else {
    weekArray[index] = date
    return fillout(weekArray, date + 1)
  }
}

// 謎の複数次元配列バグを避けるため
const removeDuplicate = (arr: any[]) => {
  let tempItem: any = null
  let tempIndex = 0
  const newArr: any[] = []
  arr.forEach((item: any, index: number) => {
    if (tempItem === item && item !== 0) {
      newArr[tempIndex] = tempItem - 1
    }
    newArr.push(item)
    tempItem = item
    tempIndex = index
  })
  return newArr
}

const Container = styled.div`
  width: 100%;
  text-align: center;
`

const YearAndMonth = styled.div`
  font-size: 40px;
  font-weight: 500;
  line-height: 1.53;
  letter-spacing: 0px;
  text-align: center;
  color: #405766;
`

const Days = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100%;
  height: 64px;
  box-sizing: border-box;
  border-bottom: 1px solid #cad5db;
`

const Day = styled.div`
  width: 40px;
  margin: 0 27px;
  font-size: 28px;
  font-weight: 500;
  text-align: center;
  color: #405766;
`

const Dates = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100%;
  height: 100px;
  box-sizing: border-box;
`

const DateElement = styled.div`
  position: relative;
  top: -12px;
  width: 50px;
  margin: 0 22px;
  font-size: 36px;
  font-weight: 500;
  letter-spacing: 0px;
  text-align: center;
  color: #405766;
`

const ReservedDot = styled.div`
  position: absolute;
  bottom: -27px;
  right: 0;
  left: 0;
  width: 16px;
  height: 16px;
  margin: auto;
  border-radius: 50%;
  background-color: #1a97fd;
`

const WaitingDot = styled.div`
  position: absolute;
  bottom: -27px;
  right: 0;
  left: 0;
  width: 16px;
  height: 16px;
  margin: auto;
  border-radius: 50%;
  &[data-request='true'] {
    background-color: #a3a3a3;
  }
  &[data-request='false'] {
    background-color: #1a97fd;
  }
`
