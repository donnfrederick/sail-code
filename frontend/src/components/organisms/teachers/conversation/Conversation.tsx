import ConversationBalloon from 'components/atoms/ConversationBalloon'
import RoundImage from 'components/atoms/RoundImage'
import NetworkStatus from 'components/atoms/teachers/NetworkStatus'
import RemainingTime from 'components/atoms/teachers/RemainingTime'
import ConversationButtons from 'components/molecules/teachers/ConversationButtons'
// import Topics from 'components/molecules/Topics'
import CloseConversationModal from 'components/organisms/teachers/modal_contents/close_conversation'
import withConversationId from 'hocs/withConversationId'
import * as ConversationModels from 'models/conversation'
import * as React from 'react'
import styled from 'styled-components'
import getPartner from 'utils/getPartner'
import resolvePath from 'utils/resolvePath'
import * as SoraClient from 'utils/SoraClient'
import Chat from 'components/molecules/Chat'
// import { translateTopicToJapanese } from 'utils/translateTopic'

interface Props {
  bandwidth: number
  chats: ConversationModels.Chats[]
  closeSoon: boolean
  conversation: ConversationModels.Conversation | null
  conversationMode: ConversationModels.ConversationMode
  currentTopic: string
  isConnecting: boolean
  myId: number
  remainingTime: string
  topics: ConversationModels.Topics
  youAreLate: boolean
  closeModal(): void
  openModal(): void
  setModalContents(contents: JSX.Element): void
}

export default withConversationId((props: Props) => {
  const {
    bandwidth,
    chats,
    closeSoon,
    conversation,
    conversationMode,
    // currentTopic,
    isConnecting,
    myId,
    remainingTime,
    // topics,
    youAreLate,
    openModal,
    setModalContents
  } = props

  if (conversation) {
    SoraClient.init(conversation.channel_id)
    SoraClient.connect()
  }

  const [isChatOpen, setIsChatOpen] = React.useState(false)
  const partner = conversation ? getPartner(conversation, myId) : null

  // const openConversationCloseModal = () => {
  //   setModalContents(<CloseConversationModal />)
  //   openModal()
  // }

  return (
    <Container>
      <RemoteVideoContainer>
        <RemoteVideo id="remote-video" autoPlay={true} playsInline={true} />
      </RemoteVideoContainer>
      <LocalVideoContainer>
        <LocalVideo id="local-video" autoPlay={true} playsInline={true} />
      </LocalVideoContainer>
      <NetworkStatus
        status={bandwidth > 300 ? 'good' : bandwidth > 100 ? 'normal' : 'bad'}
      />
      <RemainingTime remainingTime={remainingTime} />
      {conversationMode === 'soundOnly' ? (
        <Partner>
          <RoundImage
            src={
              partner
                ? partner.picture_url
                : resolvePath.image('common/user.png')
            }
            size={200}
            code={partner ? partner.country_code : ''}
          />
          <PartnerName>{partner ? partner.name : null} さん</PartnerName>
        </Partner>
      ) : null}
      {/* {isTopicsOpen ? <Topics topics={topics} /> : null}
      {!isTopicsOpen && currentTopic ? (
        <ConversationBalloon
          text={translateTopicToJapanese(currentTopic)}
          isTopic={true}
        />
      ) : null} */}
      {closeSoon ? (
        <ConversationBalloon text="そろそろお別れの時間です" />
      ) : null}
      {youAreLate ? (
        <ConversationBalloon text="遅刻ですよ！" color="#ff676e" />
      ) : null}
      {isConnecting ? <Connecting>{'接続中...'}</Connecting> : null}
      <ConversationButtons
        conversationMode={conversationMode}
        isChatOpen={isChatOpen}
        openChat={() => setIsChatOpen(!isChatOpen)}
        closeEvent={() => {
          setModalContents(<CloseConversationModal />)
          openModal()
        }}
      />
      <Canvas id="check-stream" />
      {conversation &&
        isChatOpen && (
          <ChatContainer>
            <Chat
              chats={chats}
              conversationId={conversation.id}
              openChat={() => setIsChatOpen(!isChatOpen)}
            />
          </ChatContainer>
        )}
    </Container>
  )
})

const Container = styled.div`
  position: relative;
  width: 100%;
  height: 100%;
  background-color: #000;
`

const LocalVideoContainer = styled.div`
  position: absolute;
  top: 32px;
  right: 32px;
  width: 154px;
  height: 246px;
  overflow: hidden;
`

const LocalVideo = styled.video`
  position: absolute;
  top: 50%;
  left: 50%;
  height: 100%;
  transform: translate(-50%, -50%);
`

const RemoteVideoContainer = styled.div`
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  overflow: hidden;
`

const RemoteVideo = styled.video`
  position: absolute;
  top: 50%;
  left: 50%;
  height: 100%;
  transform: translate(-50%, -50%);
`

const Partner = styled.div`
  position: absolute;
  top: 410px;
  left: 0;
  width: 100%;
  text-align: center;
`

const PartnerName = styled.div`
  margin-top: 32px;
  font-size: 48px;
  font-weight: 500;
  letter-spacing: 0px;
  text-align: center;
  color: #ffffff;
`

const Connecting = styled.div`
  position: absolute;
  top: 280px;
  right: 0;
  left: 0;
  width: 200px;
  height: 40px;
  margin: auto;
  font-size: 40px;
  color: #fff;
`

const Canvas = styled.canvas`
  display: none;
  width: 100%;
  height: 100%;
`

const ChatContainer = styled.div`
  position: absolute;
  background-color: #fff;
  bottom: 0;
  left: 0;
  width: 100%;
  height: 50%;
  overflow: hidden;
`
