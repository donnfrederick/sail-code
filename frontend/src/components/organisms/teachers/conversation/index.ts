import Conversation from 'components/organisms/teachers/conversation/Conversation'
import * as ConversationModels from 'models/conversation'
import { connect } from 'react-redux'
import { State as RootState } from 'reducers'
import * as ModalActions from 'reducers/modal'

// tslint:disable-next-line
interface OuterProps {}

interface Props {
  bandwidth: number
  chats: any[]
  closeSoon: boolean
  conversation: ConversationModels.Conversation | null
  conversationMode: ConversationModels.ConversationMode
  currentTopic: string
  isConnecting: boolean
  isTopicsOpen: boolean
  myId: number
  remainingTime: string
  topics: ConversationModels.Topics
  youAreLate: boolean
}

const mapStateToProps = (state: RootState, props: OuterProps): Props => {
  return {
    bandwidth: state.rootReducer.conversation.bandwidth,
    chats: state.rootReducer.conversation.chats,
    closeSoon: state.rootReducer.conversation.closeSoon,
    conversation: state.rootReducer.conversation.conversation,
    conversationMode: state.rootReducer.conversation.conversationMode,
    currentTopic: state.rootReducer.conversation.currentTopic,
    isConnecting: state.rootReducer.conversation.isConnecting,
    isTopicsOpen: state.rootReducer.conversation.isTopicsOpen,
    myId: state.rootReducer.teachers.me.id,
    remainingTime: state.rootReducer.conversation.remainingTime,
    topics: state.rootReducer.conversation.topics,
    youAreLate: state.rootReducer.conversation.youAreLate
  }
}

const enhancer: any = connect(
  mapStateToProps,
  {
    closeModal: ModalActions.close,
    openModal: ModalActions.open,
    setModalContents: ModalActions.setContents
  }
)

export default enhancer(Conversation)
