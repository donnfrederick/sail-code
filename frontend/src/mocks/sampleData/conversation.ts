import * as ConversationModels from 'models/conversation'

export const conversation: ConversationModels.Conversation = {
  accepting_requests: true,
  available: true,
  channel_id: '8e85f8e2-ebad-40f2-98be-b88efea45b53',
  chats: [],
  id: 26227,
  status: 'queued',
  statuses: [
    {
      user_id: 1,
      status: 'Ontime'
    },
    {
      user_id: 2,
      status: 'Ontime'
    }
  ],
  start_at: '2018-01-02T00:00:00+09:00',
  end_at: '2018-01-02T00:25:00+09:00',
  created_at: '2018-01-01T00:00:00+09:00',
  updated_at: '2018-01-01T00:00:00+09:00',
  evaluate: [
    {
      user_id: 1,
      evaluate: {
        fun: 1,
        ability: 1,
        time: 1,
        quality: []
      }
    },
    {
      user_id: 2,
      evaluate: {
        fun: 1,
        ability: 1,
        time: 1,
        quality: []
      }
    }
  ],
  memos: [
    {
      user_id: 1,
      memo: 'memo'
    },
    {
      user_id: 2,
      memo: 'memo'
    }
  ],
  reports: [
    {
      user_id: 1,
      report: {
        report_detail: 'report',
        report_reasons: [1, 2]
      }
    },
    {
      user_id: 2,
      report: {
        report_detail: 'report',
        report_reasons: [1, 2]
      }
    }
  ],
  users: [
    {
      absence: 0,
      id: 60331,
      username: 'teacher 29',
      name: '吉野 瑞葵',
      type: 'Teacher',
      sex: 2,
      picture_url: '',
      conversation_level: 5,
      level: 1,
      country: 'ja',
      country_code: 'ja',
      timezone: 'Asia/Tokyo',
      desired_condition: 2,
      evaluate: {
        1: 0,
        2: 0,
        3: 0,
        4: 0
      },
      location: '',
      lateness: 0,
      grade: null,
      highly_reliable: false,
      hobbies: [],
      payment_state: 'hidden',
      purposes: [],
      rated_conversation_level: 0,
      is_blocked: false,
      is_favorite: false,
      conversations: [],
      introduce: 'aho',
      email: ''
    },
    {
      absence: 0,
      id: 60332,
      username: 'student 29',
      name: '宮本 彩',
      type: 'Student',
      sex: 9,
      picture_url: '',
      conversation_level: 3,
      level: 3,
      country: 'TJ',
      country_code: 'TJ',
      timezone: 'Asia/Dushanbe',
      desired_condition: 1,
      evaluate: {
        1: 0,
        2: 0,
        3: 0,
        4: 0
      },
      location: '',
      lateness: 0,
      grade: 'member',
      highly_reliable: false,
      hobbies: [],
      payment_state: 'hidden',
      purposes: [],
      rated_conversation_level: 0,
      is_blocked: false,
      is_favorite: false,
      conversations: [],
      introduce: 'hage',
      email: ''
    }
  ],
  conversation_requests: []
}

export const cancelledConversation: ConversationModels.CancelledConversation = {
  conversation_id: 26227,
  start_at: '2018-01-02T00:00:00+09:00',
  end_at: '2018-01-02T00:25:00+09:00',
  created_at: '2018-01-01T00:00:00+09:00',
  updated_at: '2018-01-01T00:00:00+09:00',
  users: [
    {
      absence: 0,
      id: 60331,
      username: 'teacher 29',
      name: '吉野 瑞葵',
      type: 'Teacher',
      sex: 2,
      picture_url: '',
      level: 1,
      country: 'ja',
      country_code: 'ja',
      timezone: 'Asia/Tokyo',
      desired_condition: 2,
      evaluate: {
        1: 0,
        2: 0,
        3: 0,
        4: 0
      },
      location: '',
      lateness: 0,
      grade: null,
      highly_reliable: false,
      hobbies: [],
      payment_state: 'hidden',
      purposes: [],
      rated_conversation_level: 0,
      conversation_level: 0,
      is_blocked: false,
      is_favorite: false,
      conversations: [],
      introduce: 'cancelled_aho',
      email: ''
    },
    {
      absence: 0,
      id: 60332,
      username: 'student 29',
      name: '宮本 彩',
      type: 'Student',
      sex: 9,
      picture_url: '',
      level: 3,
      country: 'TJ',
      country_code: 'TJ',
      timezone: 'Asia/Dushanbe',
      desired_condition: 1,
      evaluate: {
        1: 0,
        2: 0,
        3: 0,
        4: 0
      },
      location: '',
      lateness: 0,
      grade: 'member',
      highly_reliable: false,
      hobbies: [],
      payment_state: 'hidden',
      purposes: [],
      rated_conversation_level: 0,
      conversation_level: 0,
      is_blocked: false,
      is_favorite: false,
      conversations: [],
      introduce: 'cancelled_hage',
      email: ''
    }
  ]
}

export const requestConversationInConversation: ConversationModels.RequestConversationInConversation = {
  end_at: '2018-01-05T23:59:59+09:00',
  id: 1,
  start_at: '2018-01-02T00:00:00+09:00',
  user: {
    absence: 0,
    id: 60331,
    username: 'teacher 29',
    name: '吉野 瑞葵',
    type: 'Teacher',
    sex: 2,
    picture_url: '',
    conversation_level: 5,
    level: 1,
    country: 'ja',
    country_code: 'ja',
    timezone: 'Asia/Tokyo',
    desired_condition: 2,
    evaluate: {
      1: 0,
      2: 0,
      3: 0,
      4: 0
    },
    location: '',
    lateness: 0,
    grade: null,
    highly_reliable: false,
    hobbies: [],
    payment_state: 'hidden',
    purposes: [],
    rated_conversation_level: 0,
    is_blocked: false,
    is_favorite: false,
    conversations: [],
    introduce: 'hagehage',
    email: ''
  }
}

export const requestConversation = {
  ...requestConversationInConversation,
  conversation
}

export const conversations: ConversationModels.ConversationResponse = {
  data: [conversation],
  meta: {
    start_on: '2018-01-02T00:00:00+09:00',
    end_on: '2018-01-05T23:59:59+09:00'
  }
}

export const requestConversations: ConversationModels.RequestConversationResponse = {
  data: [requestConversation]
}

export const teachersRequestConversations: ConversationModels.Conversation[] = [
  conversation
]

export const calendar: ConversationModels.Calendar = {
  '1': {
    is_enabled: true,
    is_reserved: false
  },
  '2': {
    is_enabled: true,
    is_reserved: false
  },
  '3': {
    is_enabled: true,
    is_reserved: false
  },
  '4': {
    is_enabled: true,
    is_reserved: false
  },
  '5': {
    is_enabled: true,
    is_reserved: false
  },
  '6': {
    is_enabled: true,
    is_reserved: false
  },
  '7': {
    is_enabled: true,
    is_reserved: false
  },
  '8': {
    is_enabled: true,
    is_reserved: false
  },
  '9': {
    is_enabled: true,
    is_reserved: false
  },
  '10': {
    is_enabled: true,
    is_reserved: false
  },
  '11': {
    is_enabled: false,
    is_reserved: true
  },
  '12': {
    is_enabled: false,
    is_reserved: true
  },
  '13': {
    is_enabled: false,
    is_reserved: true
  },
  '14': {
    is_enabled: false,
    is_reserved: true
  },
  '15': {
    is_enabled: false,
    is_reserved: true
  },
  '16': {
    is_enabled: false,
    is_reserved: true
  },
  '17': {
    is_enabled: false,
    is_reserved: true
  },
  '18': {
    is_enabled: false,
    is_reserved: true
  },
  '19': {
    is_enabled: false,
    is_reserved: true
  },
  '20': {
    is_enabled: false,
    is_reserved: true
  },
  '21': {
    is_enabled: false,
    is_reserved: false
  },
  '22': {
    is_enabled: false,
    is_reserved: false
  },
  '23': {
    is_enabled: false,
    is_reserved: false
  },
  '24': {
    is_enabled: false,
    is_reserved: false
  },
  '25': {
    is_enabled: false,
    is_reserved: false
  },
  '26': {
    is_enabled: false,
    is_reserved: false
  },
  '27': {
    is_enabled: false,
    is_reserved: false
  },
  '28': {
    is_enabled: false,
    is_reserved: false
  },
  '29': {
    is_enabled: false,
    is_reserved: false
  },
  '30': {
    is_enabled: false,
    is_reserved: false
  },
  '31': {
    is_enabled: false,
    is_reserved: false
  }
}

export const topics = {
  Traveling: '旅行',
  Reading: '読書',
  Art: '芸術'
}
