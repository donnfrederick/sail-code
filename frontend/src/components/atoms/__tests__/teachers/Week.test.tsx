import Week from 'components/atoms/teachers/Week'
import { conversations } from 'mocks/sampleData/conversation'
import * as React from 'react'
import * as renderer from 'react-test-renderer'

test('render <Week />', () => {
  const tree = renderer
    .create(
      <Week reservations={conversations} start={'2018-06-09T23:59:59+09:00'} />
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
