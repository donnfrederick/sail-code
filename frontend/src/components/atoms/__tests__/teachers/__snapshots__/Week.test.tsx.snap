// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render <Week /> 1`] = `
<div
  className="sc-bdVaJa bbgQhG"
>
  <div
    className="sc-bwzfXH cDlwhv"
  >
    2018年06月
  </div>
  <div
    className="sc-htpNat cxHgEh"
  >
    <div
      className="sc-bxivhb icxEh"
    >
      日
    </div>
    <div
      className="sc-bxivhb icxEh"
    >
      月
    </div>
    <div
      className="sc-bxivhb icxEh"
    >
      火
    </div>
    <div
      className="sc-bxivhb icxEh"
    >
      水
    </div>
    <div
      className="sc-bxivhb icxEh"
    >
      木
    </div>
    <div
      className="sc-bxivhb icxEh"
    >
      金
    </div>
    <div
      className="sc-bxivhb icxEh"
    >
      土
    </div>
  </div>
  <div
    className="sc-ifAKCX gtiVHl"
  >
    <div
      className="sc-EHOje idtMQl"
    >
      3
    </div>
    <div
      className="sc-EHOje idtMQl"
    >
      4
    </div>
    <div
      className="sc-EHOje idtMQl"
    >
      5
    </div>
    <div
      className="sc-EHOje idtMQl"
    >
      6
    </div>
    <div
      className="sc-EHOje idtMQl"
    >
      7
    </div>
    <div
      className="sc-EHOje idtMQl"
    >
      8
    </div>
    <div
      className="sc-EHOje idtMQl"
    >
      9
    </div>
  </div>
</div>
`;
